avocado/core/job.py

```diff
@@ -120,8 +120,9 @@ def __init__(self, args=None):
         #: has not been attempted.  If set to an empty list, it means that no
         #: test was found during resolution.
         self.test_suite = None
-        self.job_pre_post_dispatcher = dispatcher.JobPrePostDispatcher()
-        output.log_plugin_failures(self.job_pre_post_dispatcher.load_failures)
+
+        #: A job may not have a dispatcher for pre/post tests execution plugins
+        self.job_pre_post_dispatcher = None
```

Comment:
Sure, it's still an internal API. Anyway the sooner we start, the better. It's actually a nice custom to use private properties for internal stuff (and yes, when prototyping I tend to not use it, but when I know for sure it's not useful outside the module, I go ahead and define it as private).

Again, it was a mild suggestion.